app/controllers/credentials.controller.js

```diff
@@ -118,23 +118,25 @@ module.exports = {
     var url = connectorUrl.replace('{accountId}', accountId)
     var correlationId = req.headers[CORRELATION_HEADER] || ''
 
-    connectorClient().postAccountNotificationCredentials({
-      payload: { username, password },
-      correlationId: correlationId,
-      gatewayAccountId: accountId
-    }, function (connectorData, connectorResponse) {
-      var duration = new Date() - startTime
+    try {
+      await connectorClient().postAccountNotificationCredentials({
```

Comment:
Now that these client methods are moved over from the old base clients pub/ sub method of error propagation it would no be possible to instantiate the connector client once and then use it in the methods, this would avoid creating a more expensive object once per request. 

See https://github.com/alphagov/pay-selfservice/pull/2377/files to confirm where and when this was happening with different clients, there are some comments on the areas it was impossible to make this change before these great changes you've proposed here. 

That pattern would be something like
```js
const conncerClient = new connectorClient()

function serviceMethod() {
  return connnectorClient.postAccountNotificationCredentials({})
}
```

Reply:
Good point! I will look into instantiating client as above once connector client is fully migrated to use new baseclient.

Reply:
Changed connector client instantiation out of requests here https://github.com/alphagov/pay-selfservice/pull/2461